dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -84,6 +95,26 @@ public DhisWebServiceImpl(@Qualifier("dhisSettingsService") SettingsService sett
         this.client = httpClientBuilderFactory.newBuilder().build();
     }
 
+    @PostConstruct
+    public void discoverServerVersion() {
+        Settings settings = settingsService.getSettings();
+
+        if (StringUtils.isNotBlank(settings.getServerURI()) && StringUtils.isNotBlank(settings.getUsername()) && StringUtils.isNotBlank(settings.getPassword())) {
```

Comment:
is the check for password/user necessary? is passwordless/anonymous usage possible with DHIS?